geoips/plugin_registry.py

```diff
@@ -14,11 +14,207 @@
+            for pkg in get_entry_point_group("geoips.plugin_packages"):
+                self.registry_files.append(
+                    str(resources.files(pkg.value) / "registered_plugins.yaml")
+                )
+            self._set_class_properties()
+
+    @property
+    def registered_plugins(self):
+        """Plugin Registry registered_plugins attribute."""
+        if not hasattr(self, "_registered_plugins"):
+            self._set_class_properties()
+        return self._registered_plugins
+
+    @property
+    def interface_mapping(self):
+        """Plugin Registry interface_mapping attribute."""
+        if not hasattr(self, "_interface_mapping"):
+            self._set_class_properties()
+        return self._interface_mapping
+
+    def _set_class_properties(self):
+        """Find all plugins in registered plugin packages.
+
+        Search the ``registered_plugins.yaml`` of each registered plugin package and add
+        them to the _registered_plugins dictionary
+        """
+        # Load the registries here and return them as a dictionary
+        if not hasattr(self, "_registered_plugins"):
+            self._registered_plugins = {}
+            self._interface_mapping = {}
+            for reg_path in self.registry_files:
+                if not os.path.exists(reg_path):
+                    raise PluginRegistryError(
+                        f"Plugin registry {reg_path} did not exist, "
+                        "please run 'create_plugin_registries'"
+                    )
+                # This will include all plugins, including schemas, yaml_based,
+                # and module_based plugins.
+                pkg_plugins = yaml.safe_load(open(reg_path, "r"))
+                try:
+                    for plugin_type in pkg_plugins:
+                        if plugin_type not in self._registered_plugins:
+                            self._registered_plugins[plugin_type] = {}
+                            self._interface_mapping[plugin_type] = []
+                        for interface in pkg_plugins[plugin_type]:
+                            interface_dict = pkg_plugins[plugin_type][interface]
+                            if interface not in self._registered_plugins[plugin_type]:
+                                self._registered_plugins[plugin_type][interface] = interface_dict  # NOQA
+                                self._interface_mapping[plugin_type].append(interface)
+                            else:
+                                self.merge_nested_dicts(
+                                    self._registered_plugins[plugin_type][interface],
+                                    interface_dict,
+                                )
+                except TypeError:
+                    raise PluginRegistryError(f"Failed reading {reg_path}.")
+        return self._registered_plugins
+
+    def get_plugin_info(self, interface, plugin_name, just_info=True):
```

Comment:
This function should operate a little differently for YAML-based plugins than for module-based plugins, I think. For module-based plugins it should just return the following:
- plugin type = "module"
- interface
- family
- name
- docstring
- call signature of `call()` (if possible)

For YAML-based plugins, we can be more general, I think, and return all top-level keys that have scalar values (i.e. not mappings or lists). So, for most current plugins we would get:
- plugin_type = "yaml"
- interface
- family
- docstring
But for `product` plugins we would additionally get `source_names`.

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/pull/392/files#r1379312623
^ Is that where this was addressed?  Does "get_plugin_info" abstract the process of returning appropriate attributes / fields, and the fields are checked in "validate_plugin_info?

Reply:
I think all of your comments have been addressed. Let me push up my most recent changes to see if you agree.

Reply:
I will let @jsolbrig resolve this one.

Reply:
Evan did not include name, because it is the actual key.  Otherwise all points in here have been added to this PR.  Resolving this conversation.